### PR TITLE
TEST-#4760: Limit only ray memory to 1 GB.

### LINKED ISSUE #4760: TEST: test_io.py on ubuntu + dask gets stuck most of the time in CI
sample failure: https://github.com/modin-project/modin/runs/7640485842?check_suite_focus=true

I don't know where it's getting stuck but it's getting stuck on most CI runs.

### PR DESCRIPTION
**Pull request by mvashishtha**
## What do these changes do?

TEST-#4760: Limit only ray memory to 1 GB.

Since 53f38a4518996455b3cddaf31fd8fbf833aeee13, github CI has limited memory for
almost all tests to 1 GB, but that change was only meant for ray.

This change should not only deflake dask CI, but also speed it up significantly.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4760
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

**Review by pyrito — APPROVED** [review]
But this...does put a smile on my face

File: .github/workflows/ci.yml
Comment (vnlitvinov) [inline .github/workflows/ci.yml:518]:
```suggestion
        run: echo "MODIN_MEMORY=1000000000" >> $GITHUB_ENV
        if: matrix.engine == 'ray'
```
this looks cleaner IMHO

here and in other places

**Review by vnlitvinov — COMMENTED** [review]
I also feel you should squash the commits

**Comment by mvashishtha** [comment]
@vnlitvinov I used yaml instead of bash for the `if` as you suggested.

> I also feel you should squash the commits

Isn't that what "squash and merge" is for 😄 ?

But seriously, if there's a reason to squash commits before merge, please let me know and I can do it.

**Comment by vnlitvinov** [comment]
> Isn't that what "squash and merge" is for 😄 ?
> 
> But seriously, if there's a reason to squash commits before merge, please let me know and I can do it.

That's true, but it requires the effort to _correctly_ formulate the commit message from the one who's merging (and leaves room for error).

Other than that, no particular reason.

**Review by vnlitvinov — APPROVED** [review]
LGTM assuming tests pass.

**Pull request merged**
